test/run

```diff
@@ -153,8 +175,49 @@ def add_e2e_tests(litani, test_dir, counter, output_dir, fast):
                 ci_stage="test",
                 description=f"{test_file.stem}: transform jobs",
                 inputs=add_jobs_output,
-                phony_outputs=run_build_input,
+                phony_outputs=transform_jobs_output,
                 cwd=run_dir)
+            optional_outputs.append(transform_jobs_output)
+
+        if add_get_jobs:
+            get_jobs_output = str(uuid.uuid4())
+            litani_add(
+                litani, counter,
+                command=collapse(f"""
+                    {e2e_test_dir / 'run'}
+                    --test-file {test_file}
+                    --litani {litani}
+                    --run-dir {run_dir}
+                    --operation get-jobs"""),
+                pipeline=f"End-to-end: {test_file.stem}",
+                ci_stage="test",
+                description=f"{test_file.stem}: get jobs",
+                inputs=add_jobs_output,
+                phony_outputs=get_jobs_output,
+                cwd=run_dir)
+            optional_outputs.append(get_jobs_output)
+
+        if add_set_jobs:
+            set_jobs_output = str(uuid.uuid4())
+            litani_add(
+                litani, counter,
+                command=collapse(f"""
+                    {e2e_test_dir / 'run'}
+                    --test-file {test_file}
+                    --litani {litani}
+                    --run-dir {run_dir}
+                    --operation set-jobs"""),
+                pipeline=f"End-to-end: {test_file.stem}",
+                ci_stage="test",
+                description=f"{test_file.stem}: set jobs",
+                inputs=add_jobs_output,
+                phony_outputs=set_jobs_output,
+                cwd=run_dir)
+            optional_outputs.append(set_jobs_output)
+
+        run_build_input = add_jobs_output
```

Comment:
Should this be `run_build_input = str(uuid.uuid4())` like before?

Reply:
I think this should actually be:
1. Instead of `optional_outputs`, call it `run_build_dependencies`
2. Initially its value is `[add_jobs_output]`
3. Append the optional values as you're doing

Then just use `" ".join(run_build_dependencies)` as the input, no need for the if statement. Then run-build takes at least `add_jobs_output` as an input plus possibly some other inputs. Does that work?

Reply:
That messes with the order because we want to ensure that add_jobs runs before the "optional" outputs right? If we do it the way you're suggesting won't litani try to schedule add_jobs to run at the same time as the other tests? That means that things like transform_jobs won't have the proper jobs added to transform

Reply:
`add_jobs_output` is the output of add-jobs, so everything that takes it as an input will run after `add-jobs`. However, if there are several of them, they won't be ordered with respect to each other. So I think what I wrote is correct?

Reply:
Oh so youre suggesting that the other jobs still take add jobs as an input but the run-build one does as well? Yeah that would work, but diagram doesn't come out as clean.

Reply:
Yes, understood. Agree that the redundant edges are a bit annoying... if there's a way to have the right order without them, I'd be happy for that, but otherwise I think the edges are harmless.